integrations/unftp-sbe/src/lib.rs

```diff
@@ -14,3 +14,221 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
+
+use std::path::{Path, PathBuf};
+use std::task::Poll;
+use std::{fmt::Debug, sync::Arc};
+
+use bytes::{Buf, BufMut};
+use libunftp::auth::UserDetail;
+use libunftp::storage::{self, StorageBackend};
+use opendal::{raw::oio::PooledBuf, Buffer, Operator};
+use tokio::io::{AsyncRead, AsyncReadExt};
+
+#[derive(Debug, Clone)]
+pub struct OpendalStorage {
+    op: Operator,
+    pool: Arc<PooledBuf>,
+}
+
+impl OpendalStorage {
+    pub fn new(op: Operator) -> Self {
+        Self {
+            op,
+            pool: PooledBuf::new(16).into(),
+        }
+    }
+}
+
+/// A wrapper around [`Buffer`] to implement [`tokio::io::AsyncRead`].
+pub struct IoBuffer(Buffer);
+
+impl AsyncRead for IoBuffer {
+    fn poll_read(
+        mut self: std::pin::Pin<&mut Self>,
+        _: &mut std::task::Context<'_>,
+        buf: &mut tokio::io::ReadBuf<'_>,
+    ) -> Poll<std::io::Result<()>> {
+        let len = std::io::copy(&mut self.as_mut().0.by_ref().reader(), &mut buf.writer())?;
+        self.0.advance(len as usize);
+        Poll::Ready(Ok(()))
+    }
+}
+
+/// A wrapper around [`opendal::Metadata`] to implement [`libunftp::storage::Metadata`].
+pub struct OpendalMetadata(pub opendal::Metadata);
+
+impl storage::Metadata for OpendalMetadata {
+    fn len(&self) -> u64 {
+        self.0.content_length()
+    }
+
+    fn is_dir(&self) -> bool {
+        self.0.is_dir()
+    }
+
+    fn is_file(&self) -> bool {
+        self.0.is_file()
+    }
+
+    fn is_symlink(&self) -> bool {
+        false
+    }
+
+    fn modified(&self) -> storage::Result<std::time::SystemTime> {
+        self.0.last_modified().map(Into::into).ok_or_else(|| {
+            storage::Error::new(storage::ErrorKind::LocalError, "no last modified time")
+        })
+    }
+
+    fn gid(&self) -> u32 {
+        0
+    }
+
+    fn uid(&self) -> u32 {
+        0
+    }
+}
+
+fn cvt_err(err: opendal::Error) -> storage::Error {
```

Comment:
Please use the full name instead of abbreviations. `cvt_err` doesn't look good to me.